schema/migration.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	ljson "github.com/tigrisdata/tigris/lib/json"
+	ulog "github.com/tigrisdata/tigris/util/log"
+)
+
+// Version is schema associated with it version.
+type Version struct {
+	Version int
+	Schema  []byte
+}
+
+// Versions is an array of all collections schemas. Sorted in ascending schema version order.
+type Versions []Version
+
+func (x Versions) Len() int           { return len(x) }
+func (x Versions) Less(i, j int) bool { return x[i].Version < x[j].Version }
+func (x Versions) Swap(i, j int)      { x[i], x[j] = x[j], x[i] }
+
+func (x Versions) Latest() Version { return x[len(x)-1] }
+
+// VersionDeltaField describes field schema change.
+type VersionDeltaField struct {
```

Comment:
What if a field is dropped?

Reply:
If field is deleted then To is equal to `UnknownType`.
If field is added then From is equal to `Unknowntype`.

Reply:
added a comment.

Reply:
if a field is added would it be part of DeltaFields?

Reply:
It won't be. But just in case we need it, there is a way to express it.